### ISSUE TITLE
Flasher should write 'okok' only after verifying the flash contents. 

### ISSUE THREAD

**Issue by darconeous** [malvira/libmc1322x#4]
Currently, the flasher program simply writes the image directly to the SPI flash, starting with 'okok'. If the flash process goes bad for some reason, the chip must be erased before the flasher program can be executed again. The flasher app should be modified to write these bytes last, preferably after doing a checksum or CRC check.

**Comment by malvira** [comment]
Ack.

The whole flasher program is due for a real re-write. It would be nice to have a little protocol there where you can just manipulate the flash with serial commands. Do things like erase the whole thing, erase certain pages, write arbitrary bytes, verify blocks etc.. 

Along those lines, people are starting to use the nvm for run-time configuration now (including myself). 6lbr has https://github.com/cetic/6lbr/wiki/Nvm-Tool and I'm doing a simple structure mapped thingy in contiki. Might be nice to try to unify all of this.

I dream about having just a monolithic program that 'just do the right thing' instead of the hodgepodge of cobbled together stuff...
